### ISSUE TITLE
The previous text is displayed after changing the star rating 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#6271]
STR:
1. Load AMO dev, login and go to an add-ons detail page to make a rating
2. Click the Read all reviews link and open the reviews page 
3. Click on "Edit my review" and edit the text then click on a star to change it's rating (or the same star)
4. Observe the text

Actual result:
The previous without edits is visible.

Expected result:
The edited text is available.

Notes:
Reproducible on AMO dev with FF62 (Win10). 
Not reproducible with "Edit my review" from the add-ons detail page.

![previous text](https://user-images.githubusercontent.com/33448286/45678169-86e01900-bb3e-11e8-830a-e4dc2e18f6b4.gif)

**Comment by bobsilverberg** [comment]
This is because the review text is not saved automatically, and when the rating is changed the component reloads, losing the previously typed text. Perhaps this "review text" field is a good candidate for auto-saving? We discussed this when looking at the Edit Collection flow, noting that it can be confusing when one element is auto-saved (in this case the star rating), but another is only saved when clicking a button (in this case the review text).

Perhaps auto-saving the review text would be a good idea. What do you think @kumar303?

**Comment by kumar303** [comment]
This is a bug we should fix. When a user changes the star rating, they should not lose any text that they might have written.

**Comment by bobsilverberg** [comment]
> This is a bug we should fix. When a user changes the star rating, they should not lose any text that they might have written.

I agree, but you didn't answer my question above. We could probably fix this by making sure the whole component doesn't reload when clicking a star, but I was wondering if we should also consider doing some auto-saving for review text, which would be another way of addressing it. WDYT?

**Comment by kumar303** [comment]
There are a couple ways to fix it. I guess the easiest would be just to not reload the entire review list (which is a bit disorienting anyway). I think auto-saving is a good idea but it might take more effort. 

We actually already had this feature in the old component but it will take effort to port it to the new one since there are other components involved (like `DismissibleTextForm`).

https://github.com/mozilla/addons-frontend/blob/40d1ba4e8fe83bfdcf4bc44ba3854a5ba8a4ea61/src/amo/components/AddonReview/index.js#L103-L114

I filed this to implement auto-saving: https://github.com/mozilla/addons-frontend/issues/6348
